### PR TITLE
Add ProActive Catalog Object Dependencies Report

### PR DESCRIPTION
**Pull request by aminelouati**

### CODE REVIEW

File: src/main/java/org/ow2/proactive/catalog/callgraph/TableCallGraphsBuilder.java [hunk 1]
```diff
@@ -153,45 +155,50 @@ public void buildCallGraphsTable(CallGraphHolder globalCallGraph, BaseTable tabl
...
 
         Map<GraphNode, Graph<GraphNode, DefaultEdge>> callGraphsHashMap = new HashMap<>();
-        mergeGraphPathsHavingSameRoot.entrySet().stream().forEach(entry -> {
+        groupedGraphPathsHavingSameRoot.entrySet().stream().forEach(entry -> {
             GraphNode rootNode = entry.getKey();
             Set<GraphNode> callGraphNodes = new HashSet<>();
-            entry.getValue().stream().forEach(graphPath -> callGraphNodes.addAll(graphPath.getVertexList()));
             Set<DefaultEdge> callGraphEdges = new HashSet<>();
-            entry.getValue().stream().forEach(graphPath -> callGraphEdges.addAll(graphPath.getEdgeList()));
+            entry.getValue().stream().forEach(graphPath -> {
```
Comment (Mykhailenko) [inline src/main/java/org/ow2/proactive/catalog/callgraph/TableCallGraphsBuilder.java]:
```suggestion
            entry.getValue().forEach(graphPath -> {
```

File: src/main/java/org/ow2/proactive/catalog/report/TableCatalogObjectsDependenciesBuilder.java [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+     * This method builds a table of catalog objects dependencies ordered and grouped by bucket and object name.
+     *  In case the oder of the callGraphHolder is zero, an appropriate message is displayed.
+     * @param doc
+     * @param callGraphHolder
+     * @param catalogObjectMetadataList
+     * @param catalogObjectService
+     * @param table
+     * @throws IOException
+     */
+    public void buildCatalogObjectsDependenciesTable(PDDocument doc, CallGraphHolder callGraphHolder,
```
Comment (Mykhailenko) [inline src/main/java/org/ow2/proactive/catalog/report/TableCatalogObjectsDependenciesBuilder.java]: this method is quite long and dense. Maybe, you could find a way how to factorize it a bit? but I does not know how so i am not insisting.

File: src/main/java/org/ow2/proactive/catalog/report/TableCatalogObjectsDependenciesBuilder.java [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+            List<CatalogObjectMetadata> catalogObjectMetadataList, CatalogObjectService catalogObjectService,
+            BaseTable table) throws IOException {
+
+        if (callGraphHolder.order() == 0) {
+            Row<PDPage> dataRow = table.createRow(10f);
```
Comment (mbenguig) [inline src/main/java/org/ow2/proactive/catalog/report/TableCatalogObjectsDependenciesBuilder.java]: Define 10f as a global variable ?

File: src/main/java/org/ow2/proactive/catalog/report/TableCatalogObjectsDependenciesBuilder.java [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+            //Grouping all graph paths having the same root
+            Map<GraphNode, List<GraphPath<GraphNode, DefaultEdge>>> groupingGraphPathsHavingSameRoot = graphPathsList.stream()
+                                                                                                                     .collect(Collectors.groupingBy(GraphPath::getStartVertex,
+                                                                                                                                                    Collectors.toList()));
+
+            Set<GraphNode> leafNodes = getLeafNodes(callGraphHolder);
+            leafNodes.stream()
+                     .filter(leafNode -> !leafNode.getObjectKind().equals("N/A"))
+                     .collect(Collectors.toSet())
+                     .stream()
```
Comment (mbenguig) [inline src/main/java/org/ow2/proactive/catalog/report/TableCatalogObjectsDependenciesBuilder.java]: To be confirmed if stream() is required..

**Pull request merged**